engine/src/main/java/org/terasology/rendering/nui/widgets/UIButtonWebBrowser.java

```diff
@@ -147,16 +154,33 @@ private boolean hasConfirmation() {
     }
 
     /**
-     * Sets the {@link UIButtonWebBrowser#url} value and sets {@link UIButtonWebBrowser#confirmed}
-     * to true if the given URL is already trusted, otherwise confirmed is false.
+     * Sets the {@link UIButtonWebBrowser#url} value and sets
+     * {@link UIButtonWebBrowser#confirmed} to true if the given URL or Hostname is
+     * already trusted, otherwise confirmed is false.
+     *
+     * @throws MalformedURLException
      */
     public UIButtonWebBrowser setUrl(String url) {
+        boolean trustedHostName = false;
         boolean trustedUrl = false;
+
         if (webBrowserConfig != null) {
-            trustedUrl = webBrowserConfig.isUrlTrusted(url);
+            try {
+                String hostname = new URL(url).getHost();
+                trustedHostName = webBrowserConfig.isHostNameTrusted(hostname);
+            } catch (MalformedURLException e) {}
```

Comment:
I think we shouldn't just catch and silently ignore? Would be nice if we could somehow alert that there is a malformed URL.